FreeRTOS/Test/CMock/message_buffer/message_buffer_utest.c

```diff
@@ -0,0 +1,670 @@
+    sentBytes = xMessageBufferSend( xMessageBuffer, message, TEST_MAX_MESSAGE_SIZE, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, sentBytes );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, xStreamBufferNextMessageLengthBytes( xMessageBuffer ) );
+    TEST_ASSERT_EQUAL( 1, senderTaskWoken );
+
+    vStreamBufferDelete( xMessageBuffer );
+}
+
+/**
+ * @brief Validates a task is able to receive upto maximum message size allowed, without blocking.
+ */
+void test_xMessageBufferReceive_success( void )
+{
+    uint8_t data[ TEST_MAX_MESSAGE_SIZE ] = { 0xAA };
+    size_t sent = 0;
+    uint8_t dataReceived[ TEST_MAX_MESSAGE_SIZE + 1 ] = { 0x00 };
+    size_t received = 0;
+
+    vTaskSetTimeOutState_Ignore();
+    vTaskSuspendAll_Ignore();
+    xTaskResumeAll_IgnoreAndReturn( pdTRUE );
+
+    /* Create a message buffer of the default test sample size. */
+    xMessageBuffer = xMessageBufferCreate( TEST_MESSAGE_BUFFER_SIZE );
+    TEST_ASSERT_NOT_NULL( xMessageBuffer );
+    TEST_ASSERT_EQUAL( TEST_MESSAGE_BUFFER_SIZE, xMessageBufferSpacesAvailable( xMessageBuffer ) );
+
+    sent = xMessageBufferSend( xMessageBuffer, data, TEST_MAX_MESSAGE_SIZE, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, sent );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, xStreamBufferNextMessageLengthBytes( xMessageBuffer ) );
+
+    /* Receiving bytes of size less than the message length will return 0 bytes. */
+    received = xMessageBufferReceive( xMessageBuffer, dataReceived, TEST_MAX_MESSAGE_SIZE - 1, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    TEST_ASSERT_EQUAL( 0, received );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, xStreamBufferNextMessageLengthBytes( xMessageBuffer ) );
+
+    /* Receiving bytes of size greater than the message length will return message length bytes. */
+    received = xMessageBufferReceive( xMessageBuffer, dataReceived, TEST_MAX_MESSAGE_SIZE + 1, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, received );
+    TEST_ASSERT_EQUAL( 0, xStreamBufferNextMessageLengthBytes( xMessageBuffer ) );
+
+    vStreamBufferDelete( xMessageBuffer );
+}
+
+/**
+ * @brief Validates xMessageBufferReceive API with invalid params.
+ */
+void test_xMessageBufferReceive_invalid_params( void )
+{
+    uint8_t message[ TEST_MAX_MESSAGE_SIZE ] = { 0 };
+
+    vTaskSetTimeOutState_Ignore();
+    vTaskSuspendAll_Ignore();
+    xTaskResumeAll_IgnoreAndReturn( pdTRUE );
+
+    /* Create a message buffer of sample size. */
+    xMessageBuffer = xMessageBufferCreate( TEST_MESSAGE_BUFFER_SIZE );
+    TEST_ASSERT_NOT_NULL( xMessageBuffer );
+
+    /* Should assert if a null data buffer is passed. */
+    if( TEST_PROTECT() )
+    {
+        ( void ) xMessageBufferReceive( xMessageBuffer, NULL, TEST_MAX_MESSAGE_SIZE, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    }
+
+    validate_and_clear_assertitions();
```

Comment:
The problem is that this will cause the test to fail because the memory allocated by xMessageBufferCreate will never be freed by the call to vStreamBufferDelete.